Documentation/coding-guidelines/interop-pinvokes.md

```diff
@@ -0,0 +1,238 @@
+P/Invokes
+=========
+
+This document extends the [Interop Guidelines](interop-guidelines.md) to provide more specific guidelines, notes, and resources for defining P/Invokes.
+
+Guidelines
+----------
+1. Avoid StringBuilder
+2. Use `ExactSpelling` where possible
+3. Only use `SetLastError` where appropriate
+4. Be careful of `BOOL` vs `BOOLEAN`
+5. Watch for buffer size for output strings (null/no null)
+6. Do not create your own buffer cache, use `ArrayPool`
+7. Use blittable types in struct definitions where possible
+
+Attributes
+----------
+
+**Implicit attributes applied to parameter and return values:**
```

Comment:
We can make a consistency call here- don't use the explicit attributes unless explicitly overriding the normal implicit behavior?